### ISSUE TITLE
AASequence::fromString: increase robustness to unexpected characters

### ISSUE THREAD

**Issue by hendrikweisser** [OpenMS/OpenMS#818]
I am currently working with a FASTA file containing amino acid sequences that were translated from DNA. The sequences occasionally contain characters that represent stop codons: "*", "#", and "+".
I have been informed that this is a common occurrence in certain types of experiments/analyses. Search engines seem to be able to deal with these sequences - and so should OpenMS.  However, currently "AASequence::fromString" produces an error in such cases ("Cannot convert string into AASequence"). I believe it should instead print a warning and insert an "X" ("unknown") into the amino acid sequence.

There has been a similar problem where AASequence choked on a space in a sequence. I don't know what went wrong during the parsing of the FASTA file to create that situation, but AASequence should be able to handle it (by just skipping the space).

In conclusion, I think the string parsing functionality should be made more robust, and for unexpected characters should either insert an "X" and warn (e.g. stop codons), skip and warn (e.g. blanks), or throw an error as it does now (e.g. non-printable characters?).
